File: csharp-snippets/Main/People.cs
using System.Collections.Generic;
using Richargh.BillionDollar.Main.Repo;

namespace Richargh.BillionDollar.Main
{
    public class People : GenericRepository<PersonId, Person>
    {
        public People(IEnumerable<Person> people) : base(people)
        {
        }

        public People(params Person[] people) : base(people)
        {
        }
    }
}

File: scala-snippets/src/main/scala/de/richargh/billiondollar/People.scala
package de.richargh.billiondollar

class People(people: Person*) {
  private var allPeople: Map[PersonId, Person] = people.map(p => (p.id, p)).toMap

  def findById(id: PersonId): Option[Person] = allPeople.get(id)

  def put(person: Person): Unit = {
    allPeople += (person.id -> person)
  }
}
